static/src/js/components/CollectionSelector/__tests__/CollectionSelector.test.jsx

```diff
@@ -0,0 +1,325 @@
+              's3://example1',
+              's3://example2',
+              'prefix_bucket/*'
+            ],
+            s3CredentialsApiDocumentationUrl: 'https://example.org',
+            s3CredentialsApiEndpoint: 'https://example.org'
+          },
+          entryTitle: 'Mock title of collection 1',
+          provider: 'MMT_2',
+          shortName: 'Collection 1'
+        }
+      ])
+    })
+  })
+
+  describe('when removing all selected items', () => {
+    test('should remove all selected items and calls onChange', async () => {
+      const { user, props } = setup({})
+
+      const nameField1 = await screen.findByText('Collection 1 | Mock title of collection 1')
+      await user.click(nameField1)
+
+      const addIcon = screen.getByRole('button', { name: '+ icon' })
+
+      await user.click(addIcon)
+
+      const nameField2 = screen.getByText('Collection 2 | Mock title of collection 2')
+      await user.click(nameField2)
+
+      await user.click(addIcon)
+
+      expect(screen.getByText('Showing selected 2 items')).toBeInTheDocument()
+
+      const trashIcon = screen.getByRole('button', { name: 'trash icon' })
+      await user.click(trashIcon)
+
+      expect(screen.getByText('Showing selected 0 items')).toBeInTheDocument()
+
+      expect(props.onChange).toHaveBeenCalledWith([])
+    })
+  })
+
+  describe('when searching for available collection', () => {
+    test('should call cmr with filtered data', async () => {
+      const { user } = setup({
+        additionalMocks: [{
+          request: {
+            query: GET_PERMISSION_COLLECTIONS,
+            variables: {
+              params: {
+                keyword: 'Col',
+                limit: 20
+              }
+            }
+          },
+          result: {
+            data: {
+              collections: {
+                items: [
+                  {
+                    conceptId: 'C1200444618-MMT_2',
+                    directDistributionInformation: {
+                      region: 'us-east-2',
+                      s3BucketAndObjectPrefixNames: [
+                        's3://example1',
+                        's3://example2',
+                        'prefix_bucket/*'
+                      ],
+                      s3CredentialsApiEndpoint: 'https://example.org',
+                      s3CredentialsApiDocumentationUrl: 'https://example.org'
+                    },
+                    shortName: 'Collection 1',
+                    provider: 'MMT_2',
+                    entryTitle: 'Mock title of collection 1',
+                    __typename: 'Collection'
+                  }
+                ],
+                __typename: 'CollectionList'
+              }
+            }
+          }
+        }]
+      })
+
+      const searchField = await screen.findByPlaceholderText('Search Available...')
+
+      await user.type(searchField, 'Col')
+
+      expect(screen.getByText('Showing 1 items')).toBeInTheDocument()
```

Comment:
If this is a result of a network request happening after the user types, it might need to be `expect(await screen.findByText(....`